src/databricks/labs/ucx/source_code/notebooks/cells.py

```diff
@@ -2,12 +2,17 @@
 
 import logging
 import re
+import shlex
 from abc import ABC, abstractmethod
 from ast import parse as parse_python
 from enum import Enum
 from pathlib import Path
 from sqlglot import parse as parse_sql, ParseError as SQLParseError
 
+from pip._internal.cli.main_parser import parse_command
```

Comment:
Note this is introduces a connection with pip internals, which introduce a (higher) risk of breakage. However, I still introduced it because the reusing pips `parse_command` is preferred over reimplementing. Also, I tried to minimize the number of dependencies on pip internals.

Reply:
this means we need to bundle `pip` with the other wheels when installing in no-internet workspaces, which adds a layer of complexity. depending on `_internal` package is a slippery slope. If you want to pass down more arguments to `pip`, why not just `def register_library(self, path_lookup: PathLookup, *arguments: str)`?.. that way, both `x.register_library(lookup, "tensorflow")` and `x.register_library(lookup, "/Workspace/x/y/x.whl", "--index-url", "https://...")` work. and we don't have to reimplement anything.

Reply:
> this means we need to bundle pip with the other wheels when installing in no-internet workspaces, which adds a layer of complexity.

Isn't that also the case when using `pip` in a subprocess?

> depending on _internal package is a slippery slope

Agree!

The alternative you provided does not work well with the `Whitelist` as that requires to separate the (to-be-installed) libraries and the installation arguments. For simplification, we could only use the `Whitelist` for `len(arguments) == 0`, then the library will be installed at the cost of some overhead, but it will not be linted when an `import <library>` is defined

Reply:
> Isn't that also the case when using pip in a subprocess?

external Public API of `pip` subprocess is way more stable than any internals. e.g. what if in the next minor version they decide to move packages around?

Reply:
I removed the dependency on `pip._internal`